presto-spark-base/src/main/java/com/facebook/presto/spark/util/PrestoSparkStatsCollectionUtils.java

```diff
@@ -0,0 +1,157 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.util;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.RuntimeMetric;
+import com.facebook.presto.common.RuntimeStats;
+import com.facebook.presto.common.RuntimeUnit;
+import com.facebook.presto.execution.TaskInfo;
+import com.facebook.presto.operator.NativeExecutionInfo;
+import com.facebook.presto.operator.OperatorStats;
+import com.facebook.presto.operator.PipelineStats;
+import com.facebook.presto.operator.TaskStats;
+import org.apache.commons.text.CaseUtils;
+import org.apache.spark.executor.TaskMetrics;
+import org.apache.spark.util.AccumulatorV2;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+public class PrestoSparkStatsCollectionUtils
+{
+    public static final String SPARK_INTERNAL_ACCUMULATOR_PREFIX = "internal.metrics.";
+    public static final String PRESTO_NATIVE_OPERATOR_STATS_SEP = "root.";
+    public static final String PRESTO_NATIVE_OPERATOR_STATS_PREFIX = "velox.";
+    private static final Logger log = Logger.get(PrestoSparkStatsCollectionUtils.class);
+
+    private PrestoSparkStatsCollectionUtils() {}
+
+    public static void collectMetrics(final TaskInfo taskInfo)
+    {
+        int taskId = -1;
+        int stageId = -1;
+        try {
+            taskId = taskInfo.getTaskId().getId();
+            stageId = taskInfo.getTaskId().getStageExecutionId().getStageId().getId();
+            Set<RuntimeStats> runtimeStatsSet = collectRuntimeStats(taskInfo);
+            collectMetrics(runtimeStatsSet);
+        }
+        catch (Exception e) {
+            log.error("An error occurred while processing taskId=%s  stageId=%s", taskId, stageId, e);
+        }
+    }
+
+    public static void collectMetrics(Set<RuntimeStats> runtimeStatsSet)
+    {
+        runtimeStatsSet.forEach(runStats ->
+        {
+            runStats.getMetrics().entrySet().forEach(entry -> {
+                String prestoKey = entry.getKey();
+                String sparkInternalAccumulatorKey = getSparkInternalAccumulatorKey(prestoKey);
+                collectMetric(sparkInternalAccumulatorKey, prestoKey, entry.getValue());
+            });
+        });
+    }
+
+    static void collectMetric(final String sparkInternalAccumulatorKey,
+            final String prestoKey,
+            final RuntimeMetric metric)
+    {
+        boolean isSparkUnitMs = sparkInternalAccumulatorKey.contains("Ms");
```

Comment:
how about the metrics in `s` (second)?

Reply:
Metric unit is usually not specified on Spark side except in few cases when "Ms" is used as a prefix to indicate unit is in millisecond.